### PR TITLE
chromium: add support for external extensions

### PR DESCRIPTION
**Pull request by SebastienWae**
### Description
This allows the installation of external (outside of the Chrome Web Store) extensions, both local and remote, based on the [Alternative extension distribution options](https://developer.chrome.com/docs/extensions/mv2/external_extensions/) documentation.
Backward compatibility  for the list of string ids is maintained but I removed it from the example.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all`.

- [ ] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/doc/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

  - [ ] Added myself and the module files to `.github/CODEOWNERS`.

### CODE REVIEW

File: modules/programs/chromium.nix [hunk 1]
```diff
@@ -25,20 +25,30 @@ let
 
       extensions = mkOption {
         inherit visible;
-        type = types.listOf types.str;
+        type = types.listOf (types.either types.str
+          types.attrs); # strings are accepted to maintain backward compatibility
```
Comment (berbiche) [inline modules/programs/chromium.nix:29]: Would it be possible to change this to `types.listOf (types.coercedTo

**Review by berbiche — CHANGES_REQUESTED** [review]
Thanks for the contribution!

**Comment by SebastienWae** [comment]
@berbiche thanks for the input, let me know I need to make more changes.

Regarding the `updateUrl` type I'm using `str` instead of  `strMatching` because I could not find a regex to validate a url perfectly and handle all the possible edge cases, if someone has a better idea I'm open to suggestion.

File: modules/programs/chromium.nix [hunk 2]
```diff
@@ -25,20 +25,63 @@ let
...
+                updateUrl = mkOption {
+                  type = str;
+                  description =
+                    "URL of the extension's update manifest XML file. (Linux only)";
+                  default = "https://clients2.google.com/service/update2/crx";
+                  visible = pkgs.stdenv.isLinux;
+                  readOnly = pkgs.stdenv.isDarwin;
+                };
+                crxPath = mkOption {
+                  type = either path bool;
```
Comment (berbiche) [inline modules/programs/chromium.nix:47]: Please use `nullOr path` instead and set the default to `null`

File: modules/programs/chromium.nix [hunk 3]
```diff
@@ -25,20 +25,63 @@ let
...
+                  description =
+                    "Path to the extension's crx file. (Linux only)";
+                  default = false;
+                  visible = pkgs.stdenv.isLinux;
+                };
+                version = mkOption {
+                  type = either str bool;
```
Comment (berbiche) [inline modules/programs/chromium.nix:54]: Same as above

**Comment by rycee** [comment]
Thanks! Rebased to master in 3627ec4de58d7fbda13c82dfec94eace10198f23.

**Pull request closed without merging**